src/types/utility.ts

```diff
@@ -22,3 +22,8 @@ export type Bounds = Point & Dimensions;
 export type Matrix<T> = T[][];
 
 export type ReactHook<T, U> = (props: T) => U;
+
+/**
+ * @deprecated use Partial<Record<K, T>> directly
+ */
+export type PartialRecord<K extends keyof never, T> = Partial<Record<K, T>>;
```

Comment:
[note] This is here as a reminder/hint as to how to properly create a partial record.

Reply:
Why do we need a partial Record, if we can have `{ [key: type]: type }`?

Reply:
What does that type do, and how would it support this usecase? How does it compare/contrast to a Record?

Record<K,T> says it will contain mappings from K to T.

Partial says they may be undefined.

Reply:
off top of my head `{ [key: K]: T }` states that it is an object and if you index it with `K` it will return `T or undefined`

Reply:
Looking at the TypeScript docs, I think you're referring to 'mapped types':

- https://www.typescriptlang.org/docs/handbook/advanced-types.html#mapped-types

It basically seems like a 'type metaprogramming' method for transforming a type based on a set of 'rules'.

As examples of things that can be implemented using mapped types, they directly mention `Partial` (and some others)

We can see that `Partial` is included as a default utility type (presumably implemented on top of the basic 'mapped types' concept):

- https://www.typescriptlang.org/docs/handbook/utility-types.html

If I click through to see the implementation of `Partial` in my IDE, it looks like the following, confirming that it is just a 'wrapper' around the 'mapped types' concept:

`node_modules/typescript/lib/lib.es5.d.ts:1444`
```
/**
 * Make all properties in T optional
 */
type Partial<T> = {
    [P in keyof T]?: T[P];
};
```

---

So, long story short, `Partial` is already the canonical implementation/abstraction of what you mentioned above.

Reply:
Resolving this based on the above, if you still feel that there is a distinction/reason we should be doing it differently, feel free to re-open.